helix-term/src/ui/mod.rs

```diff
@@ -255,6 +256,54 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> Picker
     picker
 }
 
+pub fn changed_file_picker(editor: &helix_view::Editor) -> Picker<FileChange> {
+    let cwd = std::env::current_dir().unwrap_or_else(|_| PathBuf::from("./"));
+
+    let added = editor.theme.get("diff.plus");
+    let deleted = editor.theme.get("diff.minus");
+    let modified = editor.theme.get("diff.delta");
+
+    let picker = Picker::new(
+        Vec::new(),
+        menu::FileChangeData {
+            cwd: cwd.clone(),
+            style_untracked: added,
+            style_modified: modified,
+            style_deleted: deleted,
+            style_renamed: modified,
+        },
+        |cx, meta: &FileChange, action| {
+            let path_to_open = meta.path();
+            if let Err(e) = cx.editor.open(path_to_open, action) {
+                let err = if let Some(err) = e.source() {
+                    format!("{}", err)
+                } else {
+                    format!("unable to open \"{}\"", path_to_open.display())
+                };
+                cx.editor.set_error(err);
+            }
+        },
+    )
+    .with_preview(|_editor, meta| Some((meta.path().to_path_buf().into(), None)));
+    let injector = picker.injector();
+
+    let diff_providers = editor.diff_providers.clone();
+
+    std::thread::spawn(move || {
```

Comment:
Use `tokio::spawn_blocking` here. It's usually faster since it's a threadpool/reuses threads and it will make sure we have a runtime context

Reply:
Oh okay I was mimicking the `file_picker` function above actually. Lemme change this.

Reply:
Yeah i want to change that one over too. It's not a huge deal but slightly nice.